vnext/Shared/Utils.cpp

```diff
@@ -4,10 +4,49 @@
 #include "Utils.h"
 #include <regex>
 
+#include <DesktopWindowBridge.h>
+#include <ShlObj.h>
+#include <Shlwapi.h>
+#include <appmodel.h>
+#include <winrt/Windows.Storage.h>
+
 using std::string;
 
 namespace Microsoft::React {
 
+std::future<std::string> getApplicationDataPath(const wchar_t *childFolder) {
+  uint32_t length{0};
+  bool isPackagedApp = GetCurrentPackageFullName(&length, nullptr) != APPMODEL_ERROR_NO_PACKAGE;
+  std::string appDataPath;
+  if (isPackagedApp) {
+    if (!childFolder) {
+      co_return winrt::to_string(winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path());
+    } else {
+      auto subfolder = co_await winrt::Windows::Storage::ApplicationData::Current().LocalFolder().CreateFolderAsync(
+          childFolder, winrt::Windows::Storage::CreationCollisionOption::OpenIfExists);
+      co_return winrt::to_string(subfolder.Path());
+    }
+  } else {
+    wchar_t wzAppData[MAX_PATH]{};
+    if (CALL_INDIRECT(
+            L"shell32.dll", SHGetFolderPathW, nullptr, CSIDL_LOCAL_APPDATA, nullptr, SHGFP_TYPE_CURRENT, wzAppData) !=
+        S_OK) {
+      std::abort(); // We don't expect this to happen ever ..
+    }
+    appDataPath = winrt::to_string(wzAppData);
+    if (!childFolder) {
+      co_return appDataPath;
+    } else {
+      std::ostringstream os;
+      os << appDataPath << "\\" << winrt::to_string(childFolder);
+      auto childFolderPath = os.str();
+      if (!CreateDirectoryA(childFolderPath.c_str(), NULL))
```

Comment:
Since childFolderPath is a wchar, why not use `CreateDirectoryW`? Else we should probably make sure the name passed fits in the current selected OS' codePage

Reply:
As the function needs to return an 8-bit char string, we anyway have to do one conversion. So we have a choice between convert-to-8bit+process+return vs process+convert-to-8bit+return .. Processing 8bit strings should be a bit faster .. So i think we don;t gain anything by the latter .. Also, the code gets a bit clumsier when working with 16-bit string .. Hence, i'm sticking to the current flow, with a bit of cleanups after the move to new SHGetKnownFolderPath API .. Thanks !